chia/wallet/wallet_state_manager.py

```diff
@@ -660,6 +660,42 @@ async def determine_coin_type(
 
         return None, None
 
+    async def filter_spam(self, new_coin_state: List[CoinState]) -> List[CoinState]:
+        all_unspent: Set[WalletCoinRecord] = await self.coin_store.get_all_unspent_coins()
```

Comment:
I think we should first get `xch_spam_amount` from config, and if it's zero, we can avoid performing the DB lookup, and just return. This means that it will be much faster for those people that disable it, and have huge non-dust wallets

Reply:
Updated in the latest commit.